### PR TITLE
CI matrix change: add Windows.Server.Core.20H2

### PR DESCRIPTION
**Pull request by aik-jahoda**
part of #57947

OS | What to do | Notes
-- | -- | --
Windows Server Core 20H2 |  Add rolling | docker
Windows Server 20H2 | Add PR + rolling

/cc @ViktorHofer @jkotas

### CODE REVIEW

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -131,17 +131,21 @@ jobs:
...
             - Windows.10.Amd64.Server19H1.Open
+            - Windows.10.Amd64.Server20H2.Open
             - ${{ if ne(parameters.jobParameters.runtimeFlavor, 'mono') }}:
               - (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504
               - (Windows.Server.Core.1909.Amd64.Open)windows.10.amd64.server20h1.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64-20200904200251-272704c
+              - (Windows.Server.Core.20H2.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-20H2-helix-amd64-20210922171304-edc3733
           - ${{ if ne(parameters.jobParameters.isFullMatrix, true) }}:
             - Windows.81.Amd64.Open
             - Windows.10.Amd64.Server19H1.ES.Open
+            - Windows.10.Amd64.Server20H2.Open
```
Comment (jkotas) [inline eng/pipelines/libraries/helix-queues-setup.yml:142]: We need just a single Windows Servers in isFullMatrix!=true.

Comment (safern) [reply eng/pipelines/libraries/helix-queues-setup.yml:142]: I would keep the .ES. one on PRs to give us coverage for non-windows?

Comment (danmoseley) [reply eng/pipelines/libraries/helix-queues-setup.yml:142]: Makes sense to me.

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -142,6 +144,7 @@ jobs:
               - (Windows.Server.Core.1909.Amd64.Open)windows.10.amd64.server20h1.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64-20200904200251-272704c
             - ${{ if ne(parameters.jobParameters.runtimeFlavor, 'mono') }}:
               - (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504
+              - (Windows.Server.Core.20H2.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-20H2-helix-amd64-20210922171304-edc3733
```
Comment (jkotas) [inline eng/pipelines/libraries/helix-queues-setup.yml:147]: I do not think we need this for `isFullMatrix != true`

Comment (aik-jahoda) [reply eng/pipelines/libraries/helix-queues-setup.yml]: Removed 20H2 from PR run and updated table in the description.

**Review by jkotas — APPROVED** [review]
Thanks

**Pull request merged**